packages/test/test-end-to-end-tests/src/test/SummarizeFromLatest.spec.ts

```diff
@@ -0,0 +1,311 @@
+        const searchContent = await searchObj.getSearchContent();
+        if (searchContent === undefined) {
+            return undefined;
+        }
+        const content = searchContent.endsWith("\n") ? searchContent : `${searchContent}\n`;
+        return {
+            // This is the path in snapshot that ODSP expects search blob (in plain text) to be for components
+            // that want to provide search content.
+            path: ["_search", "01"],
+            content,
+        };
+    }, factory);
+}
+
+/**
+ * Validates the scenario in which we always retrieve the latest snapshot.
+ */
+describeNoCompat("Summarizer always uses the latest snapshot",
+ (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject1;
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory);
+    };
+
+    async function waitForSummary(summarizer: ISummarizer): Promise<string> {
+        // Wait for all pending ops to be processed by all clients.
+        await provider.ensureSynchronized();
+        const summaryResult = await summarizeNow(summarizer);
+        return summaryResult.summaryVersion;
+    }
+
+    beforeEach(async () => {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        mainContainer = await createContainer();
+        // Set an initial key. The Container is in read-only mode so the first op it sends will get nack'd and is
+        // re-sent. Do it here so that the extra events don't mess with rest of the test.
+        mainDataStore = await requestFluidObject<TestDataObject1>(mainContainer, "default");
+        mainDataStore._root.set("anytest", "anyvalue");
+        await waitForContainerConnection(mainContainer);
+    });
+
+    function getAndIncrementCellValue(sharedMatrix: SharedMatrix,
+        row: number, column: number, initialValue?: string): number {
+        const cellValue = sharedMatrix.getCell(row, column);
+        if (!cellValue) {
+           assert(initialValue, "Initial Value should be valid for the scenario");
+           sharedMatrix.setCell(row, column, initialValue);
+            return parseInt(initialValue, 10);
+        } else {
+            let newVal = parseInt(cellValue, 10);
+            newVal++;
+            sharedMatrix.setCell(row, column, newVal.toString());
+            return newVal;
+        }
+    }
+
+    const loadContainer = async (
+        summaryVersion: string,
+    ): Promise<IContainer> => {
+        const requestHeader = {
+            [LoaderHeader.version]: summaryVersion,
+        };
+        const container = await provider.loadContainer(runtimeFactory, undefined, requestHeader);
+        return container; // provider.loadTestContainer(undefined, requestHeader);
+    };
+
+    it("Summarizer always run from latest snapshot", async () => {
```

Comment:
I feel like the title should say something about loading summarizer from old snapshot.